api/src/main/java/uk/gov/dvsa/mot/app/MotrRequestHandler.java

```diff
@@ -0,0 +1,252 @@
+import uk.gov.dvsa.mot.vehicle.hgv.HgvVehicleProvider;
+import uk.gov.dvsa.mot.vehicle.hgv.model.HgvPsvVehicle;
+import uk.gov.dvsa.mot.vehicle.hgv.model.TestHistory;
+import uk.gov.dvsa.mot.vehicle.hgv.model.Vehicle;
+import uk.gov.dvsa.mot.vehicle.read.core.VehicleReadService;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.TemporalAdjusters;
+import java.util.Optional;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.core.Response;
+
+@Path("/")
+public class MotrRequestHandler extends AbstractRequestHandler {
+    private static final String HGV_VEHICLE_TYPE = "HGV";
+    private static final Logger logger = Logger.getLogger(MotrRequestHandler.class);
+
+    private MotrReadService motrReadService;
+    private VehicleReadService vehicleReadService;
+    private HgvVehicleProvider hgvVehicleProvider;
+    private String awsRequestId;
+
+    public MotrRequestHandler() {
+        super();
+    }
+
+    public MotrRequestHandler(boolean injectSelf) {
+        super(injectSelf);
+    }
+
+    @Inject
+    public void setVehicleReadService(VehicleReadService vehicleReadService) {
+        this.vehicleReadService = vehicleReadService;
+    }
+
+    @Inject
+    public void setHgvVehicleProvider(HgvVehicleProvider hgvVehicleProvider) {
+        this.hgvVehicleProvider = hgvVehicleProvider;
+    }
+
+    @Inject
+    public void setMotrReadService(MotrReadService motrReadService) {
+        this.motrReadService = motrReadService;
+    }
+
+    @GET
+    @Path("motr/v2/search/registration/{registration}")
+    @Produces("application/json")
+    public Response getVehicle(@PathParam("registration") String registration,
+                               ContainerRequestContext requestContext) throws TradeException {
+        try {
+
+            ApiGatewayRequestContext context = (ApiGatewayRequestContext) requestContext.getProperty(
+                    RequestReader.API_GATEWAY_CONTEXT_PROPERTY);
+            if (context != null) {
+                awsRequestId = context.getRequestId();
+            }
+
+            logger.info(String.format("Entering MotrRequestHandler, awsRequestId: %s", awsRequestId));
+
+            if (Strings.isNullOrEmpty(registration)) {
+                logger.error("Registration param is missing");
+                throw new BadRequestException("Registration param is missing", awsRequestId);
+            }
+
+            Optional<MotrResponse> motVehiclesOptional = getMotVehicle(registration);
+
+            if (motVehiclesOptional.isPresent()) {
+
+                return Response.ok(setResponseMotVehicle(motVehiclesOptional.get())).build();
+            }
+
+            Optional<DvlaVehicle> dvlaVehicleOptional = getDvlaVehicle(registration);
+
+            if (dvlaVehicleOptional.isPresent()) {
```

Comment:
Untested DVLA vehicles which are not HGVs, will not be returned from any HGV API.

Does this code expect otherwise?
Will it throw an exception for an untested car?